### ISSUE TITLE
IllegalStateException: No activity set

### ISSUE THREAD

**Issue by tasegula** [shopgun/shopgun-android-sdk#17]
We randomly get this crash:

```
Fatal Exception: java.lang.RuntimeException: Unable to pause activity {package/package.dashboard.DashboardActivity}: java.lang.IllegalStateException: No activity set in sgn:sdk:b. Make sure to instantiate ShopGun in Application.onCreate()
       at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:4046)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:4012)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3986)
       at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3960)
       at android.app.ActivityThread.-wrap15(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1624)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6710)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:770)

Caused by java.lang.IllegalStateException: No activity set in sgn:sdk:b. Make sure to instantiate ShopGun in Application.onCreate()
       at com.shopgun.android.sdk.corekit.LifecycleManager$ApplicationCallbacks.onActivityPaused(LifecycleManager.java:72)
       at android.app.Application.dispatchActivityPaused(Application.java:228)
       at android.app.Activity.onPause(Activity.java:1674)
       at android.support.v4.app.FragmentActivity.onPause(FragmentActivity.java:470)
       at com.trello.rxlifecycle2.components.support.RxAppCompatActivity.onPause(RxAppCompatActivity.java:81)
       at package.BaseActivity.onPause(BaseActivity.kt:83)
       at android.app.Activity.performPause(Activity.java:7122)
       at android.app.Instrumentation.callActivityOnPause(Instrumentation.java:1408)
       at android.app.ActivityThread.performPauseActivityIfNeeded(ActivityThread.java:4035)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:4012)
       at android.app.ActivityThread.performPauseActivity(ActivityThread.java:3986)
       at android.app.ActivityThread.handlePauseActivity(ActivityThread.java:3960)
       at android.app.ActivityThread.-wrap15(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1624)
       at android.os.Handler.dispatchMessage(Handler.java:105)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6710)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:770)
```

And this is in our `Application` class:

```
@Override
public void onCreate() {
    super.onCreate();
    ....
    new ShopGun.Builder(this).setInstance();
    ....
}
```

**Comment by tasegula** [comment]
After some investigation, we think the problem is the way `ApplicationCallbacks` is implemented.
More specifically, in `onActivityStopped` you start a delayed (3s) runnable that removes the reference for `mCurrentActivity`.
I think the problem is that another activity starts while the `Runnable` already started, so it sets the `mCurrentActivity = null`.

If I can make a suggestion, I see that `mCurrentActivity` is only used to detect if the application is active.
Maybe you use a counter that increases in `onActivityStarted` and decreses in `onActivityStopped`, so the app is active if `activityCount > 0`?

**Comment by AgneseBussone** [comment]
Fix available in 4.0.3-rc1
